### PR TITLE
OM-736 Added support for the new installation features structure

### PR DESCRIPTION
**Pull request by batmahdi**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -110,13 +110,17 @@ export class Index extends Base {
             if (this.shared.target === 'cloud') {
                 var statusData = await this.api.getStatus();
                 var featuresData = this.shared.installation.features;
+                var gateway_features = [];
```
Comment (khenderick) [inline src/index.js]: I suddenly see that this is `var`, let's use `let` instead here and on other places where `var` sneaked into the code.

File: src/index.js [hunk 2]
```diff
@@ -110,13 +110,17 @@ export class Index extends Base {
...
+                    gateway_features = gateway_features.concat(value.gateway_features);
+                  }
             } else {
                 var [statusData, featuresData] = await Promise.all([this.api.getStatus(), this.api.getFeatures()]);
             }
             if ((Toolbox.compareVersions(statusData.version, '3.143.77')) >= 0) {
                 featuresData.push('default_timer_disabled');
             }
-            this.shared.features = featuresData
+            this.shared.features = gateway_features;
```
Comment (khenderick) [inline src/index.js]: This breaks the non-cloud code.

File: src/index.js (hunk 1 above)
Comment (batmahdi) [reply src/index.js]: We can't use `let` here, `gateway_features` is used outside the if statement

Comment (batmahdi) [reply src/index.js]: It's `var` since it'll be used outside of the if statement, but will make it `let` and move it up.

File: src/index.js [hunk 3]
```diff
@@ -107,16 +107,16 @@ export class Index extends Base {
...
                 var featuresData = this.shared.installation.features;
+                for (const [_, value] of Object.entries(featuresData)) {
+                    gateway_features = gateway_features.concat(value.gateway_features);
+                }
             } else {
-                var [statusData, featuresData] = await Promise.all([this.api.getStatus(), this.api.getFeatures()]);
-            }
-            if ((Toolbox.compareVersions(statusData.version, '3.143.77')) >= 0) {
-                featuresData.push('default_timer_disabled');
+                gateway_features = await Promise.all([this.api.getStatus(), this.api.getFeatures()]);
```
Comment (khenderick) [inline src/index.js]: This won't give you the expected content in `gateway_features`

**Pull request merged**